src/puppy/platforms/win32/platform.nim

```diff
@@ -105,18 +105,77 @@ proc fetch*(req: Request): Response {.raises: [PuppyError].} =
         PuppyError, "WinHttpAddRequestHeaders error: " & $GetLastError()
       )
 
-    if WinHttpSendRequest(
-      hRequest,
-      nil,
-      0,
-      req.body.cstring,
-      req.body.len.DWORD,
-      req.body.len.DWORD,
-      0
-    ) == 0:
-      raise newException(
-        PuppyError, "WinHttpSendRequest error: " & $GetLastError()
-      )
+    if req.insecure:
+      
+      #[
+        Windows is a bit tricky with insecure ssl. Multiple requests may need to be made,
+        and such the while loop
+      ]#
+      while true:
+        if WinHttpSendRequest(
+          hRequest,
+          nil,
+          0,
+          req.body.cstring,
+          req.body.len.DWORD,
+          req.body.len.DWORD,
+          0
+        ) == 0:
+          let result = GetLastError()
+
+          #[
+            (1) If you want to allow SSL certificate errors and continue
+            with the connection, you must allow and initial failure and then
+            reset the security flags. From: "HOWTO: Handle Invalid Certificate
+            Authority Error with WinInet"
+            http://support.microsoft.com/default.aspx?scid=kb;EN-US;182888
+          ]#
+
+          if result == ERROR_WINHTTP_SECURE_FAILURE or result == ERROR_INTERNET_INVALID_CA:
+            # set the flags for insecure options
+            var dwFlags: DWORD = SECURITY_FLAG_IGNORE_UNKNOWN_CA or SECURITY_FLAG_IGNORE_CERT_WRONG_USAGE or SECURITY_FLAG_IGNORE_CERT_CN_INVALID or SECURITY_FLAG_IGNORE_CERT_DATE_INVALID
```

Comment:
Why does setting these options first does not work?

Reply:
I'm actually un aware of that my self. I've tried setting it first. Then it would crash, I guess you can't set it until you try? Windows is windows. The C++ version of this took me forever to just accept that it wont work with out it. Not without crashing.

Reply:
So basically, it's like that code that you don't know why it has to be there, but if it's not. Then stuff breaks.

Reply:
Also I could just say "cause windows"